### PR TITLE
fix: Minimize initialize at buildtime

### LINKED ISSUE #1464: Minimize build-time initialization of native-image.
## Problem
I tried using java-storage-nio on native-image (graalvm 17).
but it could not get the application default credentials at runtime.
I think this is beacuase many classes, like credentials, are initialized at build-time.

## Solution I guess
Minimize build-time initialization of native-image. (initialize only `CloudStorageFileSystemProvider` at build-time if can)
This makes it possible to get the application default credentials correctly in runtime environments.

Maybe... this problem can be solved by below simple changes I think. 

### now 
Default constructor of `CloudStorageFileSystemProvider` sets `storageOptions` to `StorageOptionsUtil.getDefaultInstance()`.
`storageOptions` are used only for late initialization of `storage`.

### changes
Default constructor of `CloudStorageFileSystemProvider` sets `storageOptions` to `null`. 
Use `StorageOptionsUtil.getDefaultInstance()` for initialize `storage` when `storageOptions` is `null`.

### PR DESCRIPTION
**Pull request by tronto20**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-storage-nio/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #1464 ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### goal
- Default constructor of `CloudStorageFileSystemProvider` references only `CloudStorageConfiguration`

### CODE REVIEW

File: google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemProvider.java [hunk 1]
```diff
@@ -1227,6 +1226,9 @@ private IOException asIoException(StorageException oops, boolean operationWasCop
 
   @VisibleForTesting
   void doInitStorage() {
-    this.storage = storageOptions.getService();
+    this.storage =
+        storageOptions != null
```
Comment (sydney-munro) [inline google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemProvider.java:1230]: why is this being moved to intialization? why can't it be left where it was?

Comment (mpeddada1) [reply google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemProvider.java:1230]: @tronto20 for visibility

Comment (tronto20) [reply google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemProvider.java:1230]: @mpeddada1 thank you

When building a graalvm native-image, `CloudStorageFileSystemProvider` and it's fields are initialized at build time.
(https://github.com/oracle/graal/issues/5134 )
`StorageOptions`, `ServiceOptions`, and their fields are also initialized.

`ServiceOptions` creates credentials, and It determines how to get application default credentials (ADC) in runtime depending on the build time environments.
(see [ServiceOptions](https://github.com/googleapis/sdk-platform-java/blob/cdcc0a201137f3cc93a96cc7670a63d7cc434142/java-core/google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java#L347))
(see [GoogleCredentialsProvider](https://github.com/googleapis/sdk-platform-java/blob/cdcc0a201137f3cc93a96cc7670a63d7cc434142/gax-java/gax/src/main/java/com/google/api/gax/core/GoogleCredentialsProvider.java#L67))

This changes makes..
1. `StorageOptions` and `GoogleCredentialsProvider` to be initialized at the runtime of actual use in graalvm native-image. (not a buildtime)
2. Applications get ADC correctly from runtime environment in graalvm native-image.

**Comment by mpeddada1** [comment]
Closing and re-opening PR to force run GA builds.

**Review by sydney-munro — APPROVED** [review]

**Pull request merged**